### ISSUE TITLE
Support for Python 3

### ISSUE THREAD

**Issue by neishm** [pygeode/pygeode#102]
This should be done in such a way that Python 2.7 is still supported (but not 2.6 or earlier?).

Our regular Python scripts can be converted with [2to3](https://docs.python.org/2/library/2to3.html), but our extension modules will be a bit tricky.  Found some info [here](http://python3porting.com/cextensions.html) which may help.

**Comment by neishm** [comment]
Pushing this to release-1.3, since it introduces a lot of changes (and potential bugs).  Will merge into master once release-1.2 is done.

**Comment by neishm** [comment]
Merged the Python3 updates into master.  Unit tests are passing for both Python 2.7 and 3.5.  Any new features from this point on should be tested against both Python2 and Python3.
